2018-05-05  Animesh Mangu 

* timespec.h (timespec_cmp): Altered function to use comparison rather
  	    		     than subtraction.

--- timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ timespecpatch.h	2018-05-05 02:46:24.350663839 -0700
@@ -45,8 +45,12 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+   if (b.tv_sec < a.tv_sec)
+      return 1;   
+   else if (b.tv_sec > a.tv_sec)
+      return -1;
+   else
+      return 0;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
